include/parameter_parser/writer/kernel.hpp

```diff
@@ -7,25 +7,56 @@
 
 namespace specfem {
 namespace runtime_configuration {
+
+/**
+ * @brief Configuration for simulation output kernel writers.
+ *
+ * Manages output format, location, and simulation type settings for
+ * kernel file generation. Creates appropriate writer instances based
+ * on configuration parameters.
+ */
 class kernel {
 public:
+  /**
+   * @brief Construct kernel configuration from explicit parameters.
+   *
+   * @param output_format File format for output (e.g., "binary", "ascii")
```

Comment:
```suggestion
   * @param output_format File format for output (e.g., "npy", "hdf5")
```
No binary option and ascii is not recommended for kernels.